compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -754,13 +805,23 @@ static LogicalResult setWarpReductionConfig(func::FuncOp entryPoint,
   }
   SmallVector<unsigned> reductionDims;
   op.getReductionDims(reductionDims);
-  if (reductionDims.size() != 1 || reductionDims[0] != op.getNumLoops() - 1)
+  if (reductionDims.empty())
     return failure();
+
+  // Make sure reduction dimensions are the innermost ones.
```

Comment:
THis is pretty complicated. Couldnt you just check this

```
int64_t numParallelDims = op.getNumParallelDims();
if (llvm::any_of(op.getReductionDims(), [&](int64_t reductionDims) { return reductionDims < numParallelDims;})) {
  return failure();
}
```

Reply:
Agreed on making it simpler. I think the check above might fail for the case of `iterator = ["reduction", "parallel", "parallel"]`.  I was thinking of using bitVector of the loop type setting parallel(1) and reduction(0), similar to how it's being used in FormDispatchRegion.cpp, but I think it may be even more complex than this current solution. 😅

Reply:
Actually, we can also use bitVector to check contiguousness, i.e `reductionLoopBitVector.find_first() - reductionLoopBitVector.find_last() == reductionLoopBitVector.size()` and another check to see if we have the last dimension ` reductionLoopBitVector.find_last() == op.getNumLoops()-1` :)

Reply:
I am confused. Dont you want it to fail for `iterator = ["reduction", "parallel", "parallel"]` . Thats what the comment says.

Reply:
ah my bad, I didn't see the `return failure();` part of the check you proposed. But wouldn't this check let `iterator = ["reduction", "reduction", "parallel", "parallel", "parallel"]` pass through?

Reply:
I dont think so... number of parallel dims is 3. IIUC `op.getReductionDims()` returns [0, 1] . So for either of those, its less than `3`, so it will return failure?

Reply:
Sorry @MaheshRavishankar, upon further investigation, you were right all along. The check you proposed should be able to handle all these cases. Updated the PR to use the check you proposed. Thanks! :)